appmonitoring/ts/src/CertificateGenerator.ts

```diff
@@ -0,0 +1,170 @@
+        }, 
+        {
+            name: 'authorityKeyIdentifier',
+            keyIdentifier: caCert.generateSubjectKeyIdentifier().getBytes(),
+        }, 
+        {
+            name: 'keyUsage',
+            digitalSignature: true,
+            keyEncipherment: true
+        },
+        {
+            name: 'extKeyUsage',
+            serverAuth: true
+        }, 
+        {
+            name: 'subjectAltName',
+            altNames: [{ type: 2, value: WebhookDNSEndpoint }]
+        }];
+
+        const newHostCert = forge.pki.createCertificate();
+        const hostKeys = forge.pki.rsa.generateKeyPair(4096);
+
+        // Set the attributes for the new Host Certificate
+        newHostCert.publicKey = hostKeys.publicKey;
+        newHostCert.serialNumber = CertificateManager.randomHexSerialNumber();
+        newHostCert.validity.notBefore = new Date(timeNowNum - (5 * 60 * 1000)); //5 Mins ago
+        newHostCert.validity.notAfter = new Date(timeNowNum + (2 * 365 * 24 * 60 * 60 * 1000)); //2 Years from now
+        newHostCert.setSubject(host_attributes);
+        newHostCert.setIssuer(caCert.subject.attributes);
+        newHostCert.setExtensions(host_extensions);
+
+        // Sign the new Host Certificate using the CA
+        newHostCert.sign(caCert.privateKey, forge.md.sha256.create());
+
+        // // Convert to PEM format
+        const pemHostCert = forge.pki.certificateToPem(newHostCert);
+        const pemHostKey = forge.pki.privateKeyToPem(hostKeys.privateKey);
+
+        return {
+            caCert: caCertResult.certificate,
+            tlsCert: pemHostCert,
+            tlsKey: pemHostKey
+        } as WebhookCertData;
+    }
+
+    public static async PatchSecretStore(kubeConfig: k8s.KubeConfig, certificate: WebhookCertData) {
+        const secretsApi = kubeConfig.makeApiClient(k8s.CoreV1Api);
+        const secretStore = await secretsApi.readNamespacedSecret(CertificateStoreName, NameSpaceName);
+        const secretsObj: k8s.V1Secret = secretStore.body;
+
+        secretsObj.data['ca.cert'] = btoa(certificate.caCert);
+        secretsObj.data['tls.cert'] = btoa(certificate.tlsCert);
+        secretsObj.data['tls.key'] = btoa(certificate.tlsKey);
+
+        await secretsApi.patchNamespacedSecret(CertificateStoreName, NameSpaceName, secretsObj, undefined, undefined, undefined, undefined, undefined, {
+            headers: { 'Content-Type' : 'application/strategic-merge-patch+json' }
+        });
+    }
+
+    public static async PatchMutatingWebhook(kubeConfig: k8s.KubeConfig, certificate: WebhookCertData) {
+        const webhookApi: k8s.AdmissionregistrationV1Api = kubeConfig.makeApiClient(k8s.AdmissionregistrationV1Api);
+        const mutatingWebhook = await webhookApi.readMutatingWebhookConfiguration(WebhookName);
+        const mutatingWebhookObject: k8s.V1MutatingWebhookConfiguration = mutatingWebhook.body;
+        mutatingWebhookObject.webhooks[0].clientConfig.caBundle = btoa(certificate.caCert);
```

Comment:
Let's check that webhooks isn't empty and throw a meaningful message if it is. Can't just hope for the best, need to be defensive.